ci/arrow-build.sh

```diff
@@ -32,9 +37,9 @@ cmake -DARROW_BUILD_TESTS=FALSE \
       -DARROW_FILESYSTEM=ON \
       -DARROW_JSON=ON \
       -DARROW_PARQUET=ON \
-      ..
+      "$SRC_DIR"
 
-make arrow
+make -j 4 arrow
```

Comment:
`-j 4` ^_^

Reply:
Oh I guess, should this be an environment variable provided by GitHub actions or do we know it's usually 4?

Reply:
Yeah on a second thought probably makes sense to replace this with `$(nproc)` instead, which would be slightly more future-proof :)